js/geo/transform.js

```diff
@@ -260,6 +260,16 @@ Transform.prototype = {
     },
 
     /**
+     * Like `locationPoint` but returns a raw 4d vector containing info about the transform
+     * @param {LngLat} lnglat location
+     * @returns {Array} The 4d vector pertaining to this point's transform
+     * @private
+     */
+    locationPoint3d: function(lnglat) {
+        return this.coordinatePoint3d(this.locationCoordinate(lnglat));
+    },
```

Comment:
I'm pretty sure using 3d points would be a major undertaking across the codebase.

Right now these points don't actually represent a the position of the point on the map; just the position of the point on the screen. Everything is being projected down to 2d. Trying to match the proper 3d to 2d mapping with CSS using an x-y-z point instead would be hard.